fix(sync): partial result should continue to reconcile

jrcichra opened the pull request:
When there is a partial failure, lastErr was being set. This caused the reconciler to stop reconciling conditions.

We should surface partial failures but not in a way that stops the reconciler from applying the partial conditions.

When there is a partial result, omit the errors but still surface the issue in the get_failures metric.

muff1nman commented on internal/alert/sync.go:
```diff
@@ -280,6 +278,12 @@ func (p *PromMultiClient) GetAlerts(ctx context.Context) ([]promv1.Alert, bool,
+	// do not return errors on partial failures
+	// metrics will surface the issue
+	// and conditions will still reconcile
+	if partial {
```
should paritial distinguish between all of the sources failing and at least one source failing?

jrcichra replied:
That sounds reasonable, we can do a calculation here to determine if it's truly partial or not.

Commit:
fix(sync): partial result should continue to reconcile

When there is a partial failure, lastErr was being set. This caused the
reconciler to stop reconciling conditions.

We should surface partial failures but not in a way that stops the
reconciler from applying the partial conditions.

When there is a partial result, omit the errors but still surface the
issue in the get_failures metric.
